key vault 'NestedItemType' should be one of: keys, secrets, certificates, got "storage"

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.5.3

### AzureRM Provider Version

3.63.0

### Affected Resource(s)/Data Source(s)

azurerm_key_vault_managed_storage_account

### Terraform Configuration Files

```hcl
resource "azurerm_key_vault_managed_storage_account" "sa" {
  depends_on                   = [azurerm_key_vault.kv, azurerm_key_vault_access_policy.terraform_access_policy]
  count                        = var.sas_enabled ? 1 : 0
  name                         = var.storage
  key_vault_id                 = azurerm_key_vault.kv.id
  storage_account_id           = var.storage_id
  storage_account_key          = "key1"
  regenerate_key_automatically = true
  regeneration_period          = "P90D"
}
```

### Debug Output/Panic Output

```shell
##[error]Terraform command 'plan' failed with exit code '1'.
##[error]╷
│ Error: key vault 'NestedItemType' should be one of: keys, secrets, certificates, got "storage"
│ 
│   with module.basis.module.altinn-keyvault.azurerm_key_vault_managed_storage_account.sa[0],
│   on .terraform/modules/basis/modules/aks-keyvault/storage-sas-token.tf line 1, in resource "azurerm_key_vault_managed_storage_account" "sa":
│    1: resource "azurerm_key_vault_managed_storage_account" "sa" {
```

### Expected Behaviour

Terraform should run through with result no change.

### Actual Behaviour

Terraform plan fail.

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

Regression from 3.62.0 to 3.63.0

https://github.com/hashicorp/terraform-provider-azurerm/compare/v3.62.0...v3.63.0

I guess this section should include 'storage'

"internal/services/keyvault/parse/nested_item.go"
```
type NestedItemObjectType string

const (
	// KeyVaultObjectType Keys...
	NestedItemTypeKey NestedItemObjectType = "keys"
	// KeyVaultObjectType Secrets...
	NestedItemTypeSecret NestedItemObjectType = "secrets"
	// KeyVaultObjectType Certificates...
	NestedItemTypeCertificate NestedItemObjectType = "certificates"
)
```

--- a/internal/services/keyvault/parse/nested_item.go
+++ b/internal/services/keyvault/parse/nested_item.go
@@ -22,11 +22,13 @@ const (
 	NestedItemTypeSecret NestedItemObjectType = "secrets"
 	// KeyVaultObjectType Certificates...
 	NestedItemTypeCertificate NestedItemObjectType = "certificates"
+	// KeyVaultObjectType Storage Keys...
+	NestedItemTypeStorageKey NestedItemObjectType = "storage"
 )
 
 // PossibleNestedItemObjectTypeValues returns a string slice of possible "NestedItemObjectType" values.
 func PossibleNestedItemObjectTypeValues() []string {
-	return []string{string(NestedItemTypeKey), string(NestedItemTypeSecret), string(NestedItemTypeCertificate)}
+	return []string{string(NestedItemTypeKey), string(NestedItemTypeSecret), string(NestedItemTypeCertificate), string(NestedItemTypeStorageKey)}
 }
 
 var _ resourceids.Id = NestedItemId{}

